src/arch/linux/mapping/mapping.c

```diff
@@ -398,26 +398,11 @@ static int mapping_is_hole(void *start, size_t size)
   return (mapping_find_hole(beg, end, size) == start);
 }
 
-static void *do_mmap_mapping(int cap, void *target, size_t mapsize, int protect)
+static void *do_mmap_mapping(int cap, dosaddr_t targ, size_t mapsize, int protect)
 {
-  void *addr;
-  int flags = (target != (void *)-1) ? MAP_FIXED : 0;
-  /* assume fixed mappings are already visible to kvm unless marked immediate */
-  int update_kvm = ((target == (void *)-1) || (cap & MAPPING_IMMEDIATE));
-
-  if (target == (void *)-1) target = NULL;
-#ifdef __x86_64__
-  if (flags == 0 &&
-      (cap & (MAPPING_DPMI|MAPPING_VGAEMU|MAPPING_INIT_LOWRAM|MAPPING_KVM)))
-    flags = _MAP_32BIT;
```

Comment:
Should we remove MAP_32BIT?

Reply:
That's #1844. I'm not sure why that's still stuck?

Reply:
Is there any in particular? I will double check but I think I had resolved/answered everything.

Yes, we do need MAP_32BIT for native DPMI. That's done correctly in #1844

Reply:
https://github.com/dosemu2/dosemu2/pull/1844#discussion_r1056624797
https://github.com/dosemu2/dosemu2/pull/1844#discussion_r1056625766